### DISCUSSION TITLE
cann't retrieve process.env by destructuring assignment, eg process.env.NEXT_PUBLIC_ENV

### DISCUSSION THREAD

**Discussion by dotku** [vercel/next.js#68241] [Help]
### Summary

I'm trying to create something easy access env by `const NEXT_PUBLIC_ENV=process.env` but failed :(

### Additional information

```js
process.env.NEXT_PUBLIC_ENV prints but destructuring assignment failed


const NEXT_PUBLIC_ENV=process.env
console.log(NEXT_PUBLIC_ENV)
```

May I know why?
```


### Example

https://github.com/dotku/zyintl/blob/5ef02b7209cc4d7766f591590d55013cdf6817c0/src/app/auth/page.jsx

**Comment by Juneezee** [comment]
This is actually intended and not a bug. See https://nextjs.org/docs/app/api-reference/next-config-js/env

> Next.js will replace process.env.customKey with 'my-value' at build time. Trying to destructure process.env variables won't work due to the nature of webpack [DefinePlugin](https://webpack.js.org/plugins/define-plugin/).
